pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/SourceIdentifierScanner.scala

```diff
@@ -50,19 +50,24 @@ object SourceIdentifierScanner extends Logging {
     root.sourceIdentifier.json
       .getOption(node)
       .map(parseSourceIdentifier)
-      .map { sourceIdentifier =>
-        identifiers.get(sourceIdentifier) match {
-          case Some(Identifier(canonicalId, _, _, _)) =>
-            root.obj.modify { identifierJson =>
-              ("canonicalId", Json.fromString(canonicalId)) +: identifierJson
-            }(node)
-          case None =>
-            throw new RuntimeException(
-              s"Could not find $sourceIdentifier in $identifiers")
-        }
+      .map(getCanonicalId(identifiers))
+      .map { canonicalId =>
+        root.obj.modify { obj =>
+          ("canonicalId", Json.fromString(canonicalId)) +: obj
+        }(node)
       }
       .getOrElse(node)
 
+  private def getCanonicalId(identifiers: Map[SourceIdentifier, Identifier])(
+    sourceIdentifier: SourceIdentifier): String =
+    identifiers
```

Comment:
This is definitely clearer.  You could avoid the exception here by dealing with `Either` everywhere  but that might be overcomplicating things.